### PR TITLE
Autotools: Implement symbols visibility control

### PR DESCRIPTION
**Pull request by evpobr**
* Use symbol file under Win32 with MinGW only
* Use unified visibility control for other platforms
* Add recommended `win32-dll` option to `LT_INIT` macro

### CODE REVIEW

File: src/libsndfile-1.def [hunk 1]
```diff
@@ -1,6 +1,4 @@
-; Auto-generated by cmake
-
-LIBRARY ${PROJECT_NAME}-${PROJECT_VERSION_MAJOR}.dll
+LIBRARY libsndfile-1.dll
```
Comment (erikd) [inline src/libsndfile-1.def]: Is `LIBRARY ${PROJECT_NAME}-${PROJECT_VERSION_MAJOR}.dll` not correct? I think it is. I don't see any reason to make this change (or the similar one above).

Comment (evpobr) [inline src/libsndfile-1.def]: It was correct.

I thought that this file is changing very rarely, so there is not much point in configuring it every time.

But of course i can make a single template for an Autotools and CMake and configure it with something like `PACKAGE` and `VERSION_MAJOR`.

I know how to do this with CMake, but can you give me `VERSION_MAJOR` regex? I see `VERSION_MINOR` in `configure.ac`, something like that. I'm not very friendly with regular expressions.

Comment (erikd) [inline src/libsndfile-1.def]:
```
VERSION_MAJOR=`echo $PACKAGE_VERSION | $SED "s/\..*//"`
```

Comment (evpobr) [inline src/libsndfile-1.def]: @erikd:

        ; Auto-generated by @GEN_TOOL@

`GEN_TOOL` for CMake is set to `cmake`. `autoconf` for Autotools is OK?

Comment (erikd) [inline src/libsndfile-1.def]: Sounds reasonable.

**Comment by erikd** [comment]
Should we wait for fixed Appveyor builds for this?

**Comment by evpobr** [comment]
Small fix just to remove AppVeyor test results. I guess we can merge it right now.

**Pull request merged**